packages/ceramic-core/src/document.ts

```diff
@@ -677,15 +692,15 @@ class Document extends EventEmitter {
    * Gracefully closes the document instance.
    */
   async close (): Promise<void> {
-    this.off('update', this._handleTip.bind(this))
+    this.off('update', this._update.bind(this))
```

Comment:
I don't think that works, every time the function is bound it's going to return a new bound function, so the callback attached is not actually removed. 
Instead the `_update` method itself could be bound, ex:
```ts
_update = async (cid: CID): Promise<void> => {
  try {
    await this._handleTip(cid)
  } catch (e) {
    this._logger.error(e)
  } finally {
    this._isProcessing = false
  }
}
```
and
```ts
this.on('update', this._update)
this.off('update', this._update)
```

Reply:
This is the old code so I didn't want to go deep into. So, what you're saying is to remove `.bind(this)` for `on` and `off`, both?

Reply:
@PaulLeCam removed binding 👍